src/components/signUp/Form.tsx

```diff
@@ -0,0 +1,128 @@
+import axios, { AxiosError } from 'axios';
+import { useState } from 'react';
+import 'components/signUp/Form.css'
+
+const Form = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordConfirmation, setpasswordConfirmation] = useState("");
+  const [name, setName] = useState("");
+  const [gender, setGender] = useState("");
+  const [requestError, setrequestError] = useState(false);
+
+  async function SignUp() {
+    debugger;
+    try {
+      const res = await axios.post(
+        'http://localhost:3001/api/v1/users',
+        { user:
+          {
+            email: email,
+            password: password,
+            password_confirmation: passwordConfirmation,
+            gender: gender,
+            username: name
+          }
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json'
+          }}
+        );
+      
+      setrequestError(false);
+      return {
+        accessToken: res.headers['access-token'],
+        client: res.headers['client'],
+        uid: res.headers['uid']
+      }
+    }
+    catch(e: any){
+      debugger;
+      setrequestError(true);
+    }
+  }
+
+  return (
+    <form
+      onSubmit={e => {
+      e.preventDefault();
+      SignUp();
+    }}>
+
+      {requestError === true?
+        <p className="signup-error">An error has occurred</p>: null
+      }
```

Comment:
You can make this line shorter doing this:
```suggestion
      {requestError && <p className="signup-error">An error has occurred</p>}
```